examples/compositor.rs

```diff
@@ -63,192 +57,21 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
 struct ClientState;
 impl ClientData<App> for ClientState {
-    fn initialized(&self, client_id: ClientId) {
+    fn initialized(&self, _client_id: ClientId) {
         println!("initialized");
     }
 
-    fn disconnected(&self, client_id: ClientId, reason: DisconnectReason) {
+    fn disconnected(&self, _client_id: ClientId, _reason: DisconnectReason) {
         println!("disconnected");
     }
 }
 
-impl GlobalDispatch<WlCompositor> for App {
-    type GlobalData = ();
-
-    fn bind(
-        &mut self,
-        handle: &mut wayland_server::DisplayHandle<'_, Self>,
-        client: &wayland_server::Client,
-        resource: New<WlCompositor>,
-        global_data: &Self::GlobalData,
-        data_init: &mut DataInit<'_, Self>,
-    ) {
-        DelegateGlobalDispatch::<WlCompositor, _>::bind(
-            &mut CompositorDispatch(&mut self.compositor_state, &mut self.inner),
-            handle,
-            client,
-            resource,
-            global_data,
-            data_init,
-        );
-    }
-}
+delegate_global_dispatch!(App: [WlCompositor] => CompositorState);
+delegate_dispatch!(App: [WlCompositor] => CompositorState);
+delegate_dispatch!(App: [WlSurface] => CompositorState);
+delegate_dispatch!(App: [WlRegion] => CompositorState);
+delegate_dispatch!(App: [WlCallback] => CompositorState);
```

Comment:
FYI you should be able to combine the various delegates into a single macro call:
```suggestion
delegate_dispatch!(App: [WlCompositor, WlSurface, WlRegion, WlCallback] => CompositorState);
```

Reply:
Wow, that's cool!